### ISSUE TITLE
Promtail deployment yaml template

### ISSUE THREAD

**Issue by nickbp** [opstrace/opstrace#688]
Similar to #687 except for deploying promtail. Relatively lower priority but doesn't hurt to try.

Main tasks are:
- testing/validation of the template across a couple clusters to check that it actually deploys/works
- At the moment the template assumes dockerd is being used (looks for logs in /var/log/docker/...), need to see whether we can get it to work with containerd. Can check promtail docs for any tips.

**Comment by nickbp** [comment]
This seems to do the job, logs are a bit different where it doesn't look like there's an equivalent to the metric name needed:

query for a single log line with matching integration_id over the last hour, see also [docs](https://grafana.com/docs/loki/latest/api/#examples-1), e.g. docs say timestamps are epoch nanoseconds and default to `start=now-1h`, `end=now`
```
GET https://default.MYCLUSTER.opstrace.io/grafana/api/datasources/proxy/2/loki/api/v1/query_range?direction=BACKWARD&limit=1&query={integration_id="INTEGRATION_ID_HERE"}&start=1620865766000000000&end=1620869367000000000&step=2
```

**Comment by nickbp** [comment]
Per comment in the above commit, the promtail config needs the user to specify whether their cluster is running dockerd or cri/containerd. This is mainly because the logs produced by each have totally different formats and we need to specify which log parser promtail should be using against the log files. If it weren't for this then we probably could've gotten away with a common universal config, but them's the breaks.

The config itself seems to be fully functional on a test EKS(dockerd) cluster, while an onprem(containerd) cluster has some labeling/pod discovery issues (but is successfully scraping the promtail pods themselves). So at this point there may just need to be some tweaks to the relabeling config and then we should be good to go.

**Comment by nickbp** [comment]
Turns out the only issue was with a relabeling config, where promtail was expected pods to be labeled a certain way (and was ignoring pods that weren't). I've cleaned up the relabeling rules across metrics/prometheus and logs/promtail integrations, and tested both across an eks/dockerd cluster and an onprem/containerd cluster:
- I checked the prometheus/promtail detection UI wasn't dropping anything unexpected. It looked like everything was being discovered correctly for both metrics and logs.Similarly I checked Opstrace Grafana and was able to find logs/metrics for all of them as well.

**Comment by nickbp** [comment]
The promtail YAML definition is looking good per above, and there's now a basic dashboard created for the integration as well.

Due to the nature of how the logs work, there's a lot less structure there compared to metrics, and a lot of the dashboards for logs effectively involve defining regexps against the log data itself. As a result the initial sample dashboard is mainly just showing the amount of data received across various dimensions, without trying to do much filtering against it. A way that we could improve this in the future would be to prompt the user for some things they want to query - I've filed #732 for tracking this.

In the meantime, though, I've been able to deploy the promtail configuration in both dockerd and cni/containerd modes across an EKS (dockerd) cluster and an onprem/K3s (containerd) cluster. In both cases the sample dashboard was populated with results that appeared to make sense. I've thrown these over the fence into the `i9n/k8s-logs` branch.
